File: testing/ChromaClient.cpp
//
//
#include "zhelpers.hpp"
#include <string>

static zmq::socket_t * s_client_socket (zmq::context_t & context) {
  zmq::socket_t * client = new zmq::socket_t (context, ZMQ_REP);
  client->connect ("tcp://localhost:5556");
  return client;
} 
int main () {
  zmq::context_t context(1);
 
  zmq::socket_t *client = s_client_socket (context);

  while (true) {
    std::string msg = s_recv (*client);
    if (msg == "go"){
      s_send (*client, "");
    }
    bool expect_reply = true;
    while (expect_reply) {
      std::string data = s_recv (*client);
      std::cout << data << "\n" << "sending back";
      s_send (*client, data);
      expect_reply = false;
      //can expand this loop to implement some cheap reliability
   }
  }
  delete client;
  return 0;
}

File: testing/ChromaClient.py
import sys
import time
import zmq

context = zmq.Context()
socket = context.socket(zmq.REP)
socket.connect("tcp://localhost:5556")

def GetData():
    data = socket.recv()
    print "received data %s"%(data)
    return data
def doSomething(data1):
    reversed1 = data1[::-1]
    return reversed1
def sendBack(data1):
    print "sending data back"
    signal = socket.recv()
    if signal == "go":
        socket.send(data1)
def sendSignal(somestring):
    socket.send(somestring)
def waitForSignal():
    msg = socket.recv()
    return msg

def main():
    if waitForSignal() is not None:
        sendSignal("")
        jobID = GetData()
        sendSignal("")
        sendBack(doSomething(jobID))
        exit(1)
    else:
        main()


if __name__ == "__main__":
    main()
        
        
    

